c/tskit/tables.c

```diff
@@ -10301,6 +10428,74 @@ tsk_table_collection_load_indexes(tsk_table_collection_t *self, kastore_t *store
     return ret;
 }
 
+static int
+tsk_reference_sequence_load(tsk_reference_sequence_t **self, kastore_t *store)
```

Comment:
I'm finding this a bit convoluted. How about we make this a method of the table collection, like 
```
tsk_table_collection_load_reference_sequence...
```
then later
```c
reference_sequence_loaded = data != NULL || url != NULL || metadata != NULL || metadata_schema != NULL;
if (self->reference_sequence != NULL) {
     free
}
if (reference_sequence_loaded) {
    self->reference_sequence = malloc
     if (data != NULL) {
         ret = tsk_reference_sequence_set_data(*self, data, (tsk_size_t) data_length);  
         if (ret != 0) {
             goto out;
         }
     }
     // etc
    }
```
I think this captures the various permutations of reference/no reference before/after ok, right?

Reply:
Done